interp/implicit_quantifiers.ml

```diff
@@ -92,9 +92,6 @@ let free_vars_of_constr_expr c ?(bound=Id.Set.empty) l =
   let rec aux bdvars l c = match CAst.(c.v) with
     | CRef (qid,_) when qualid_is_ident qid ->
       found c.CAst.loc (qualid_basename qid) bdvars l
-    | CNotation (_,(InConstrEntry,"{ _ : _ | _ }"), ({ CAst.v = CRef (qid,_) } :: _, [], [], [])) when
-        qualid_is_ident qid && not (Id.Set.mem (qualid_basename qid) bdvars) ->
-        Constrexpr_ops.fold_constr_expr_with_binders (fun a l -> Id.Set.add a l) aux (Id.Set.add (qualid_basename qid) bdvars) l c
```

Comment:
Good question. I suspect that my reasoning was that we are now able to know in `Constrexpr_ops.fold_constr_expr_with_binders` what terms are binders in the substitution of the notation.

In the general case, there was also the open problem that a notation does not keep the information of what what subexpression binds what other subexpression. For that purpose, I added that information in the `Notation_term.NtnTypeVar` (see the comment at line 68 of `notation_term.mli`).

I don't remember if I completed the work at propagating this information. This could be checked.